File: 2640-maximum-number-of-integers-to-choose-from-a-range-i/2640-maximum-number-of-integers-to-choose-from-a-range-i.java
import java.util.HashSet;
import java.util.Arrays;
import java.util.List;
import java.util.ArrayList;
import java.util.Collections;

class Solution {
    public int maxCount(int[] banned, int n, int maxSum) {
        HashSet<Integer> numberSet = new HashSet();
        for (int i = 1; i <= n; i += 1) {
            numberSet.add(i);
        }
        for (int b: banned) {
            numberSet.remove(b);
        }
        List<Integer> numbers = new ArrayList(numberSet);
        Collections.sort(numbers);

        int count = 0;
        int sum = 0;
        for (int num: numbers) {
            sum += num;
            if (sum > maxSum) {
                return count;
            }
            count += 1;
        }
        return count;
    }
}

File: 2640-maximum-number-of-integers-to-choose-from-a-range-i/2640-maximum-number-of-integers-to-choose-from-a-range-i.scala

object Solution {
    def maxCount(banned: Array[Int], n: Int, maxSum: Int): Int = {
        val bannedSet = banned.toSet
        val numbers = (1 to n).filter(!bannedSet.contains(_))
        numbers.zipWithIndex
               .foldLeft(Left(0).asInstanceOf[Either[Int, Int]]) {
            case (Left(sum), (cur, id)) =>
                if sum + cur > maxSum then Right(id)
                else Left(sum + cur)
            case (right, _) => right
        } match {
            case Left(_) => numbers.length
            case Right(len) => len
        }
    }
}